sn_client/src/wallet.rs

```diff
@@ -74,6 +74,18 @@ impl WalletClient {
         }
     }
 
+    /// Get storecost from the network
+    /// this uses the largest DBC as the payment section
+    pub async fn get_store_cost(&mut self) -> Result<Token> {
+        // For now we simply just use the largest DBC we have...
+        let (dbc, _key) = self.wallet.largest_dbc()?;
```

Comment:
Not sure about this `using the largest dbc's cost as the cost of the whole wallet` idea
this `get_store_cost` function is for the whole wallet, indicating the wallet is stored as a whole.
However, as I understand, there is no such `whole wallet stored in network` , instead, there is just a bunch of DBCs scattered across the network?
In that case, given DBC is also kind of small fixed sized, shall we just charge a fixed cost for each storage?
then the `cost of whole wallet` just being a calculation of : `num_of_dbc * fixed_price`

Reply:
> Not sure about this using the largest dbc's cost as the cost of the whole wallet idea

That's fair, and is largely just placeholder to get us going. It doesn't hold up for any payment across more than one dbc eg. I'm hoping this bad impl spurs us on to better things ;)

> instead, there is just a bunch of DBCs scattered across the network

yup


But the cost isn't about the DBC, it's about what _some relevant nodes_ think the price is _right now_. 

Here i chose `NetworkAddress<DbcId>` nodes as I thought they were validating things. But they are not! it's all in `kad::put`. So it's the nodes storing records that validate. So price could be determined from... anywhere? every chunk? I'm not sure what makes sense as a best effort there. But I think we can vastly improve this in follow ups for sure!

Reply:
Right, so if currently we just to make a generic approach of `how to decide storege cost`, then I think the targeted address shall be as generic as `NetworkAddress`, instead of a specific dbc_id.
And further to that, the cost is calculated by the close group of the address.

I think the part of using `the medium cost collected from close group` is OK,
however targeting `highest dbc's id` is bit suspicious to me.

Reply:
> then I think the targeted address
What's the `targeted address` here? (atm, I think _any_ query is better that the status quo, so unless it's a simple change I'd perhaps opt for a follow up?)

Or do you literally just mean switch the api interface to `NetworkAddress` there?

>  however targeting highest dbc's id is bit suspicious to me.

How so? (and would this a blocker or something to follow up with for you?)

Reply:
> How so? (and would this a blocker or something to follow up with for you?)

Regarding the current `get_store_cost` function API, it seems assuming the cost of storage is `unified` acorss the network.
Which not reflecting the network reality, but may be a `convenient assumption for simpler implementation` ? 
Further to that, as we already assumed the charge will be `unified`, then actually doesn't matter using `largest dbc` or not. Just query any random section will do the same.

> Or do you literally just mean switch the api interface to NetworkAddress there?

Yes, and that `NetworkAddress` is from `address of the data to be stored`

Reply:
>  Which not reflecting the network reality, but may be a convenient assumption for simpler implementation ?
💯 

>Further to that, as we already assumed the charge will be unified, then actually doesn't matter using largest dbc or not. Just query any random section will do the same.

Not 100% in this impl. As this should still get us the _verifying_ nodes just now. As it's those that'll handle the DBC (which is why we use the dbc_id). (Which comes into play with the follow up node-verification pr)


In the end we could just sample the network and go off of that, most likely.


> Yes, and that NetworkAddress is from address of the data to be stored

Considering the above I wouldn't do that in this version. I definitely agree there are improvements to be made.